rpcs3/Emu/RSX/VK/vkutils/device.h

```diff
@@ -132,37 +133,38 @@ namespace vk
-		bool get_anisotropic_filtering_support() const;
-		bool get_wide_lines_support() const;
-		bool get_conditional_render_support() const;
-		bool get_unrestricted_depth_range_support() const;
-		bool get_external_memory_host_support() const;
-		bool get_surface_capabilities_2_support() const;
-		bool get_debug_utils_support() const;
-		bool get_descriptor_indexing_support() const;
-
-		u64 get_descriptor_update_after_bind_support() const;
-		u32 get_descriptor_max_draw_calls() const;
-
-		VkQueue get_present_queue() const;
-		VkQueue get_graphics_queue() const;
-		VkQueue get_transfer_queue() const;
-		u32 get_graphics_queue_family() const;
-		u32 get_present_queue_family() const;
-		u32 get_transfer_queue_family() const;
-
-		mem_allocator_base* get_allocator() const;
-
-		operator VkDevice() const;
+		const physical_device& gpu() const { return *pgpu; }
+		const memory_type_mapping& get_memory_mapping() const { return memory_map; }
+		const gpu_formats_support& get_formats_support() const { return m_formats_support; }
+		const pipeline_binding_table& get_pipeline_binding_table() const { return m_pipeline_binding_table; }
+		const gpu_shader_types_support& get_shader_types_support() const { return pgpu->shader_types_support; }
+
+		bool get_shader_stencil_export_support() const { return pgpu->stencil_export_support; }
+		bool get_depth_bounds_support() const { return pgpu->features.depthBounds != VK_FALSE; }
+		bool get_alpha_to_one_support() const { return pgpu->features.alphaToOne != VK_FALSE; }
+		bool get_anisotropic_filtering_support() const { return pgpu->features.samplerAnisotropy != VK_FALSE; }
+		bool get_wide_lines_support() const { return pgpu->features.wideLines != VK_FALSE; }
+		bool get_conditional_render_support() const { return pgpu->conditional_render_support; }
+		bool get_unrestricted_depth_range_support() const { return pgpu->unrestricted_depth_range_support; }
+		bool get_external_memory_host_support() const { return pgpu->external_memory_host_support; }
+		bool get_surface_capabilities_2_support() const { return pgpu->surface_capabilities_2_support; }
+		bool get_debug_utils_support() const { return g_cfg.video.renderdoc_compatiblity && pgpu->debug_utils_support; }
+		bool get_descriptor_indexing_support() const { return pgpu->descriptor_indexing_support; }
+		bool get_framebuffer_loops_support() const { return pgpu->framebuffer_loops_support; }
+
+		u64 get_descriptor_update_after_bind_support() const { return pgpu->descriptor_update_after_bind_mask; }
+		u32 get_descriptor_max_draw_calls() const { return pgpu->descriptor_max_draw_calls; }
+
+		VkQueue get_present_queue() const { return m_present_queue; }
+		VkQueue get_graphics_queue() const { return m_graphics_queue; }
+		VkQueue get_transfer_queue() const { return m_transfer_queue; }
+		u32 get_graphics_queue_family() const { return m_graphics_queue_family; }
+		u32 get_present_queue_family() const { return m_graphics_queue_family; }
```

Comment:
typo?

Reply:
No, we don't support a standalone present queue at the moment. Present and graphics share a queue.